File: src/main/java/io/github/marcuscastelo/invtweaks/client/behavior/IInvTweaksBehavior.java
package io.github.marcuscastelo.invtweaks.client.behavior;

import io.github.marcuscastelo.invtweaks.operation.OperationInfo;
import io.github.marcuscastelo.invtweaks.operation.OperationResult;

public interface IInvTweaksBehavior {
    OperationResult sort(OperationInfo operationInfo);
    OperationResult moveAll(OperationInfo operationInfo);
    OperationResult dropAll(OperationInfo operationInfo);
    OperationResult moveAllSameType(OperationInfo operationInfo);
    OperationResult dropAllSameType(OperationInfo operationInfo);
    OperationResult moveOne(OperationInfo operationInfo);
    OperationResult dropOne(OperationInfo operationInfo);
    OperationResult moveStack(OperationInfo operationInfo);
    OperationResult dropStack(OperationInfo operationInfo);
    OperationResult craftOne(OperationInfo operationInfo);
    OperationResult craftStack(OperationInfo operationInfo);
    OperationResult craftAll(OperationInfo operationInfo);
    OperationResult craftAllSameType(OperationInfo operationInfo);
}

File: src/main/kotlin/io/github/marcuscastelo/invtweaks/behavior/IInvTweaksBehavior.kt
package io.github.marcuscastelo.invtweaks.behavior

import io.github.marcuscastelo.invtweaks.operation.OperationInfo
import io.github.marcuscastelo.invtweaks.operation.OperationResult

interface IInvTweaksBehavior {
    fun sort(operationInfo: OperationInfo): OperationResult
    fun moveAll(operationInfo: OperationInfo): OperationResult
    fun dropAll(operationInfo: OperationInfo): OperationResult
    fun moveAllSameType(operationInfo: OperationInfo): OperationResult
    fun dropAllSameType(operationInfo: OperationInfo): OperationResult
    fun moveOne(operationInfo: OperationInfo): OperationResult
    fun dropOne(operationInfo: OperationInfo): OperationResult
    fun moveStack(operationInfo: OperationInfo): OperationResult
    fun dropStack(operationInfo: OperationInfo): OperationResult
    fun craftOne(operationInfo: OperationInfo?): OperationResult
    fun craftStack(operationInfo: OperationInfo?): OperationResult
    fun craftAll(operationInfo: OperationInfo?): OperationResult
    fun craftAllSameType(operationInfo: OperationInfo?): OperationResult
}